Owin.Security.Providers/Foursquare/FoursquareAuthenticationHandler.cs

```diff
@@ -0,0 +1,243 @@
+					return new AuthenticationTicket(null, properties);
+				}
+
+				var graphResponse = await this._httpClient.GetAsync(GraphApiEndpoint + "?oauth_token=" + Uri.EscapeDataString(accessToken) + "&m=foursquare&v=" + DateTime.Today.ToString("yyyyyMMdd"), this.Request.CallCancelled);
+				graphResponse.EnsureSuccessStatusCode();
+
+				var accountstring = await graphResponse.Content.ReadAsStringAsync();
+				var accountInformation = JObject.Parse(accountstring);
+				var user = (JObject)accountInformation["response"]["user"];
+
+				var context = new FoursquareAuthenticatedContext(this.Context, user, accessToken);
+
+				context.Identity = new ClaimsIdentity(
+					new[]
+					{
+						new Claim(ClaimTypes.NameIdentifier, context.Id, XmlSchemaString, this.Options.AuthenticationType),
+						new Claim(ClaimTypes.Name, context.Name, XmlSchemaString, this.Options.AuthenticationType),
+						new Claim("urn:foursquare:id", context.Id, XmlSchemaString, this.Options.AuthenticationType),
+						new Claim("urn:foursquare:name", context.Name, XmlSchemaString, this.Options.AuthenticationType),
+					},
+					this.Options.AuthenticationType,
+					ClaimsIdentity.DefaultNameClaimType,
+					ClaimsIdentity.DefaultRoleClaimType);
+
+				if (string.IsNullOrWhiteSpace(context.Email) == false)
+				{
+					context.Identity.AddClaim(new Claim(ClaimTypes.Email, context.Email, XmlSchemaString, this.Options.AuthenticationType));
+				}
+
+				if (string.IsNullOrWhiteSpace(context.Twitter) == false)
+				{
+					context.Identity.AddClaim(new Claim("urn:foursquare:twitter", context.Twitter, XmlSchemaString, this.Options.AuthenticationType));
+				}
+
+				await this.Options.Provider.Authenticated(context);
+
+				context.Properties = properties;
+
+				return new AuthenticationTicket(context.Identity, context.Properties);
+			}
+			catch (Exception ex)
+			{
+				this._logger.WriteWarning("Authentication failed", ex);
+				return new AuthenticationTicket(null, properties);
+			}
+		}
+
+		protected override Task ApplyResponseChallengeAsync()
+		{
+			this._logger.WriteVerbose("ApplyResponseChallenge");
+
+			if (this.Response.StatusCode != (int)HttpStatusCode.Unauthorized)
+			{
+				return Task.FromResult<Object>(null);
+			}
+
+			var challenge = Helper.LookupChallenge(this.Options.AuthenticationType, this.Options.AuthenticationMode);
+
+			if (challenge != null)
+			{
+				var baseUri = this.Request.Scheme + Uri.SchemeDelimiter + this.Request.Host + this.Request.PathBase;
+				var currentUri = baseUri + this.Request.Path + this.Request.QueryString;
+				var redirectUri = baseUri + this.Options.CallbackPath;
+
+				var extra = challenge.Properties;
+
+				if (string.IsNullOrEmpty(extra.RedirectUri) == true)
+				{
+					extra.RedirectUri = currentUri;
+				}
+
+				// OAuth2 10.12 CSRF
+				this.GenerateCorrelationId(extra);
+
+				// OAuth2 3.3 space separated
+				var scope = string.Join(" ", this.Options.Scope);
```

Comment:
This variable is not assigned. If it is not needed can you please remove it and the `Scope` properties from the options class. If it is needed, please update the query string. Thank you.